File: web/src/lib/languages.ts
export type Language = {
  code: string;
  label: string;
};

export const LANGUAGES: Language[] = [
  { code: "en", label: "English" },
  { code: "fa", label: "Persian" },
  { code: "ar", label: "Arabic" },
  { code: "tr", label: "Turkish" },
  { code: "de", label: "German" },
  { code: "fr", label: "French" },
  { code: "es", label: "Spanish" },
  { code: "ru", label: "Russian" },
];

export function languageLabel(code: string): string {
  return LANGUAGES.find((l) => l.code === code)?.label ?? code.toUpperCase();
}

File: mobile/lib/core/languages.dart
class Language {
  const Language(this.code, this.label);
  final String code;
  final String label;
}

const supportedLanguages = <Language>[
  Language('en', 'English'),
  Language('fa', 'Persian'),
  Language('ar', 'Arabic'),
  Language('tr', 'Turkish'),
  Language('de', 'German'),
  Language('fr', 'French'),
  Language('es', 'Spanish'),
  Language('ru', 'Russian'),
];

String languageLabel(String code) {
  for (final lang in supportedLanguages) {
    if (lang.code == code) return lang.label;
  }
  return code.toUpperCase();
}
